unit-tests/src/test/scala/java/lang/ThreadTest.scala

```diff
@@ -0,0 +1,64 @@
+// Ported, with thanks & gratitude, from Scala.js
+// 2020-09-20
+// Scala.js Repository Info
+//   commit: 9dc4d5b36ff2b2a3dfe2e91d5c6b1ef6d10d3e51
+//   commit date: 2018-10-11
+//
+// Slightly modified for Scala Native.
+
+/*
+ * Scala.js (https://www.scala-js.org/)
+ *
+ * Copyright EPFL.
+ *
+ * Licensed under Apache License 2.0
+ * (https://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package java.lang
+
+import org.junit.Test
+import org.junit.Assert._
+
+class ThreadTest {
+
+  val executingInJVM = false
+
+  @Test def getName_and_setName(): Unit = {
+    if (!executingInJVM) {
+      val t = Thread.currentThread()
+      assertEquals("main", t.getName) // default name of the main thread
+      t.setName("foo")
+      try {
+        assertEquals("foo", t.getName)
+      } finally {
+        t.setName("main") // don't pollute the rest of the world with this test
+      }
+      assertEquals("main", t.getName)
+    }
+  }
+
+  @Test def currentThread_getStackTrace(): Unit = {
+    val trace = Thread.currentThread().getStackTrace()
+    assertEquals(trace.length, 0)
```

Comment:
Done.  Adding the hard coded variable ` val executingInScalaNative = true`
is not particularly elegant but I could find no relevant Platform.scala or PlatformSpecific.scala.  Perhaps in the future. ScalaNative is not set up as
a cross-project.

 There is a Platform.scala, but that is used for something entirely else.

I can change the name if recommended (or do something else entirely). 
I do think it is worth having the test, even though it will need to be changed
when/if the stub is replaced by a proper implementation.

Reply:
In the rest of Scala.js tests, we use `!executingInJVM` for that purpose, rather than defining another `executingInJS`. Perhaps you can use the same so that it is coherent by anticipation with other tests we might port from Scala.js in the future?

Reply:
I considered `!executingInJVM` as your comment above recommends. The problem with that that
some keen & eagle-eyed reviewer is likely to ask, as they should, "What happens when this code
executes in JS?".  The assertion will fail because this clause is really only Scala-Native, because of 
the stub.   

I think the "at least think unified & portable code base" is a wise & essential way to go.

In the interest of efficient & effective use of time all around, I propose that I drop the
Scala-Native-only zero-element result test.   It will just have to be removed anyway
in the future when someone implements the method being tested.

Yes, this kicks both the "do a proper implementation of the method" and "Provide
a proper PlatformSpecific.scala" down the road. 
   
I am open to other possibilities.

Please advise.

Reply:
We can keep it as is for now. It's good to at least call the method to prove that it links.

Reply:
Agreed, thank you for merge. 

As I poke around, especially with testing, I will keep the need for something like "PlatformSpecific", which
at least provides a "executingInJVM", etc. in mind.   

I know about the System.getProperty("mumble.vm.name") test, the that seems to be a pretty long code path.
I dislike proposing/introducing dependencies on long code paths, especially runtime.  I think the "executingInJVM"
test should be close to the metal/compiler.

I will try to create an Issue to capture this concern.  Progress comes in fits & starts.